primer/brand
# Fix missing anchor nav active link on page load

## Summary

Fixes issue where the active link isn't visible on initial mount.

Resolves https://github.com/primer/brand/issues/319
Closes https://github.com/primer/brand/pull/332 

## List of notable changes:



- Fixed the bug by adding a default value if the active link is initially undefined
- Updated rootMargin and threshold values for when Intersection Observer deems the element `isIntersecting`.
- Added new story controls to increase/decrease the height of anchored sections (useful for testing)

## What should reviewers focus on?



- Verify the fix works for you

## Steps to test:



1. [Open this story to the reproducible bug in production right now](https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-anchornav-features--narrow-view&args=offset:500)
1. [Open this story in the preview deployment where it's fixed](https://primer-f744f3a55a-26139705.drafts.github.io/storybook/?path=/story/components-anchornav-features--narrow-view&args=offset:500)
1. Verify the active link is now showing

## Supporting resources (related issues, external links, etc):

- [Original bug report](https://github.com/primer/brand/issues/319)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">


![Screenshot 2023-10-05 at 15 05 09](https://github.com/primer/brand/assets/13340707/bf2dccb5-cdae-4748-809f-ff2182f7f1fb)

 </td>
<td valign="top">

![Screenshot 2023-10-05 at 15 05 20](https://github.com/primer/brand/assets/13340707/da7e8c0d-1cb0-4284-87ec-da43ff944097)


</td>
</tr>
</table>
